crypto/crypto.go

```diff
@@ -27,18 +26,23 @@ func Digest(ms ...[]byte) []byte {
 	return ret
 }
 
-func GenerateKey() KeyPair {
-	pk, sk, _ := ed25519.GenerateKey(rand.Reader)
-	return KeyPair{
-		PrivateKey: sk,
-		PublicKey:  pk,
-	}
+func GenerateKey() (SigningKey, error) {
+	_, sk, err := ed25519.GenerateKey(rand.Reader)
+	return SigningKey(sk), err
 }
 
-func Sign(key KeyPair, message []byte) []byte {
-	return ed25519.Sign(key.PrivateKey, message)
+func (s SigningKey) public() crypto.PublicKey {
+	return ed25519.PrivateKey(s).Public()
 }
 
-func Verify(key KeyPair, message, sig []byte) bool {
-	return ed25519.Verify(key.PublicKey, message, sig)
+func Sign(key SigningKey, message []byte) []byte {
+	return ed25519.Sign(ed25519.PrivateKey(key), message)
+}
+
+func Verify(key SigningKey, message, sig []byte) bool {
```

Comment:
I would understand the need for the `public()` method if you were passing in a `crypto.PublicKey` here, but otherwise it's probably unnecessary.  The internals of `Verify()` are already implementation specific.

Also, why the local typedef for `SigningKey` but the imported `crypto.PublicKey`? I imagine disambiguating `crypto` in other parts of the codebase (should it be used) may be cumbersome, so probably best to avoid that if possible.